src/sfptpd_clock.c

```diff
@@ -2484,4 +2571,163 @@ bool sfptpd_clock_is_active(const struct sfptpd_clock *clock)
 	return clock && !clock->deleted;
 }
 
+int sfptpd_clock_deduplicate(void)
+{
+	struct sfptpd_clock **snapshot;
+	struct sfptpd_clock *sys;
+	size_t count = 0;
+	/* A large settlement period is required; at least >1s in case of PPS
```

Comment:
We should probably note this in the release notes.  A delay on startup is fine as long as the customer knows to expect it.  A "We used your option and now it's slow to start up" ticket is much less fun.

Reply:
Yes - assuming you wish this to be a supported feature. I've added a note to the changelog.